src/phpDebug.ts

```diff
@@ -264,8 +273,30 @@ class PhpDebugSession extends vscode.DebugSession {
             });
             server.listen(args.port || 9000, (error: NodeJS.ErrnoException) => error ? reject(error) : resolve());
         });
+
+        /** sets up a client to register IDE to DBgp proxy */
+        const createProxyClient = () => new Promise((resolve, reject) => {
+            const client = this._proxyClient = net.createConnection({port: args.proxyPort | 9001, host: args.proxyHost}, () => {
+                this.sendEvent(new vscode.OutputEvent('DBgp proxy: connected to ' + args.proxyHost + ':' + args.proxyPort + '\n'));
+                client.write('proxyinit -p ' + args.proxyPort + ' -k ' + args.ideKey + ' -m 1');
+            });
+
+            client.on('data', (data:string) => {
+                client.end();
```

Comment:
You just end the socket on the first chunk of data, but you don't check/parse the content. This means you don't handle errors - the proxy will respond with a `proxyinit` package that can contain an error:
```xml
<proxyinit success="[0|1]"
           idekey="{ID}"
           address="{IP_ADDRESS}"
           port="{NUM}">
    <error id="app_specific_error_code">
        <message>UI Usable Message</message>
    </error>
</proxyinit>
```
The whole proxy init should not happen over a raw net socket. Instead, all the command-queue logic should be moved from `XDebugConnection` to `DBGpConnection`, and then there should be a new class `DBGpProxyConnection` that has a `sendProxyInitCommand` and `sendProxyStopCommand` method. The parsing is already implemented, that way you reduce the code here and can handle the error easily.

The `proxystop` command is currently not implemented at all.